Added From for BufferedDecoder.

Razican opened the pull request:
Reduced memory footprint by using `Box<[u8]>` instead of `Vec<u8>`, and fixed some non-idiomatic details here and there. Formatted the code, but still some `rustfmt` warnings left.

gnieto commented:
Thanks  @Razican !

I broke your branch as I've just merged a PR that removed some code that you formatted. Can you rebase it, please?

Also, I'll add `rustfmt` checks on travis to avoid the formatting issues on PRs.

Razican committed:
Added From for BufferedDecoder.

Reduced memory footprint by using Box<[u8]> instead of Vec<u8>, and
fixed some non-idiomatic details here and there. Formatted the code,
but still some rustfmt warnings left.

Razican commented:
Should be good now

Razican committed:
Fixed tests
